lusidtools/cocoon/cocoon_printer.py

```diff
@@ -0,0 +1,171 @@
+
+    return (pd.DataFrame(items_success, columns=["successful items"]),
+            errors)
+
+
+def format_transactions_response(response: dict) -> (pd.DataFrame, pd.DataFrame):
+    """
+    This function unpacks a response from transaction requests and returns successful and errored statuses for
+    request constituents.
+    :param dict response: response from Lusid-python-tools
+    :param str file_type: file_type of Lusid-python-tools
+    :return: pd.DataFrame: Successful calls from request
+    :return: pd.DataFrame: Error responses from request that fail (APIExceptions: 400 errors)
+    """
+    file_type = "transactions"
+
+    check_dict_for_required_keys(response[file_type], f"Response from {file_type} request", ["errors", "success"])
+
+    # get success
+    items_success = [batch.href for batch in response[file_type]["success"]]
+
+    errors = get_errors_from_response(response[file_type]["errors"])
+
+    return (pd.DataFrame(get_portfolio_from_href(items_success, file_type), columns=["successful items"]),
+            errors)
+
+
+def format_holdings_response(response: dict) -> (pd.DataFrame, pd.DataFrame):
+    """
+    This function unpacks a response from holding requests and returns successful and errored statuses for
+    request constituents.
+    :param dict response: response from Lusid-python-tools
+    :return: pd.DataFrame: Successful calls from request
+    :return: pd.DataFrame: Error responses from request that fail (APIExceptions: 400 errors)
+    """
+    file_type = "holdings"
+    check_dict_for_required_keys(response[file_type], f"Response from {file_type} request", ["errors", "success"])
+
+    # get success
+    items_success = [batch.href for batch in response[file_type]["success"]]
+
+    errors = get_errors_from_response(response[file_type]["errors"])
+
+    return (pd.DataFrame(get_portfolio_from_href(items_success, file_type), columns=["successful items"]),
+            errors)
+
+
+def format_quotes_response(response: dict) -> (pd.DataFrame, pd.DataFrame, pd.DataFrame):
+    """
+    This function unpacks a response from quotes requests and returns successful, failed and errored statuses for
+    request constituents.
+    :param dict response: response from Lusid-python-tools
+    :return: pd.DataFrame: Successful calls from request
+    :return: pd.DataFrame: Error responses from request that fail (APIExceptions: 400 errors)
+    :return: pd.DataFrame: Failed responses that LUSID rejected
+    """
+    file_type = "quotes"
+
+    check_dict_for_required_keys(response[file_type], f"Response from {file_type} request", ["errors", "success"])
+
+    items_success = [batch.values[key].quote_id.quote_series_id.instrument_id for batch in
+                     response[file_type]["success"] for key in batch.values.keys()]
+
+    items_failed = [batch.failed[key].quote_id.quote_series_id.instrument_id for batch in
+                    response[file_type]["success"] for key in batch.failed.keys()]
+
```

Comment:
Make this more unique by creating a key which uses other fields from the quote id. As you may have multiple quotes for a given instrument ID.